### PR TITLE
Make more of lab manager dependencies optional

### PR DESCRIPTION
**Pull request by vidartf**
It would make the lab manager more reusable if it had fewer strict requirements. I kept the rendermime dependency strict, as I see no way to use it without this.

Other things that I added since they were related (see individual commits for split):
- Cleaned up use of types.
- Fixed an assign in `if` clause (should have been compare `===`).
- Added strict null checks to lab manager package.

If any of these auxiliary points are contentious at all, just say the word and I will split up the PR.

### CODE REVIEW

File: packages/jupyterlab-manager/src/plugin.ts [hunk 1]
```diff
@@ -168,17 +168,37 @@ function updateSettings(settings: ISettingRegistry.ISettings) {
...
-  });
+  if (settingRegistry !== null) {
+    settingRegistry.load(plugin.id).then((settings: ISettingRegistry.ISettings) => {
+      settings.changed.connect(updateSettings);
+      updateSettings(settings);
+    }).catch((reason: Error) => {
+      console.error(reason.message);
+    });
+
+    // Add a command for creating a new Markdown file.
```
Comment (jasongrout) [inline packages/jupyterlab-manager/src/plugin.ts]: This comment apparently was a copy-paste error originally.

File: packages/jupyterlab-manager/src/manager.ts [hunk 2]
```diff
@@ -266,8 +266,8 @@ class WidgetManager extends ManagerBase<Widget> implements IDisposable {
    * Get the currently-registered comms.
    */
   async _get_comm_info(): Promise<any> {
-    const reply = await this._context.session.kernel.requestCommInfo({target: this.comm_target_name});
-    if (reply.content.status = 'ok') {
+    const reply = await this._context!.session.kernel!.requestCommInfo({target: this.comm_target_name});
+    if (reply.content.status === 'ok') {
```
Comment (jasongrout) [inline packages/jupyterlab-manager/src/manager.ts]: Thanks for catching this!

**Comment by jasongrout** [comment]
At first glance, this looks great!

**Comment by vidartf** [comment]
Let's wait on a rebase after #1734 to make this diff smaller...

**Comment by vidartf** [comment]
I cleaned up this PR a lot, so it should now be a lot quicker to review!

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**